ArcadiaTeamsBot/Infrastructure/Bot.cs

```diff
@@ -1,32 +1,50 @@
 ﻿namespace ArcadiaTeamsBot.Infrastructure
 {
+    using System;
     using System.Collections.Generic;
     using System.Threading;
     using System.Threading.Tasks;
 
     using Microsoft.Bot.Builder;
+    using Microsoft.Bot.Builder.Dialogs;
     using Microsoft.Bot.Schema;
 
-    public class Bot : ActivityHandler
+    public class Bot<T> : ActivityHandler where T : Dialog
     {
-        protected override async Task OnMembersAddedAsync(IList<ChannelAccount> membersAdded, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+        protected readonly Dialog Dialog;
+        protected readonly BotState ConversationState;
+
+        private const string WelcomeText = "Welcome to ArcadiaBot, {0}.  You can create a new request or view opened requests.";

```

Comment:
Here too. Constants should be placed prior to fields.

Reply:
Replaced